### PR TITLE
[wip] Darwin fixes

### PR DESCRIPTION
**Pull request by matthewbauer**
###### Things done:
- [ ] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- [ ] Built on platform(s): NixOS / OSX / Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: pkgs/games/scummvm/default.nix [hunk 1]
```diff
@@ -2,13 +2,14 @@
...
+
   src = fetchurl {
     url = "mirror://sourceforge/scummvm/${name}.tar.bz2";
     sha256 = "d9ff0e8cf911afa466d5456d28fef692a17d47ddecfd428bf2fef591237c2e66";
   };
-  
-  buildInputs = [ SDL zlib libmpeg2 libmad libogg libvorbis flac alsaLib ];
+
+  buildInputs = [ SDL zlib libmpeg2 libmad libogg libvorbis flac ]
+    ++ stdenv.optionals stdenv.isLinux [ alsaLib ];
```
Comment (puffnfresh) [inline pkgs/games/scummvm/default.nix]: stdenv.lib.optionals

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -12963,6 +12963,8 @@ let
   mplayer = callPackage ../applications/video/mplayer ({
     pulseSupport = config.pulseaudio or false;
     libdvdnav = libdvdnav_4_2_1;
+    inherit (darwin) IOKit;
+    inherit (darwin.apple_sdk.frameworks) Carbon Cocoa;
```
Comment (puffnfresh) [inline pkgs/top-level/all-packages.nix]: What happens when accessing these attributes on non-Darwin platforms?

File: pkgs/applications/video/mplayer/default.nix [hunk 3]
```diff
@@ -20,6 +20,9 @@
 , jackaudioSupport ? false, libjack2 ? null
 , pulseSupport ? false, libpulseaudio ? null
 , bs2bSupport ? false, libbs2b ? null
+, fbdevSupport ? stdenv.isLinux
+, vidixSupport ? !stdenv.isDarwin
+, IOKit, Carbon, Cocoa
```
Comment (puffnfresh) [inline pkgs/applications/video/mplayer/default.nix]: Seems like these should have default values of null.

**Comment by vbgl** [comment]
Maybe you should update the `meta.platforms` attributes of `gnubg` and `mplayer` to reflect that they are supported on darwin.

When trying to build `mplayer` on mac osx 10.11.3, I get the following error:

```
In file included from input/ar.c:26:
In file included from /nix/store/dfvrzc0p99vp3gx2q7w79iiq14drj851-IOKit-osx-10.10.5/Library/Frameworks/IOKit.framework/Headers/hid/IOHIDLib.h:31:
/nix/store/dfvrzc0p99vp3gx2q7w79iiq14drj851-IOKit-osx-10.10.5/Library/Frameworks/IOKit.framework/Headers/hid/IOHIDDevice.h:315:1: error: expected function body after function declarator
AVAILABLE_MAC_OS_X_VERSION_10_10_AND_LATER;
^
```

I have no clue what that means.

**Comment by matthewbauer** [comment]
@vbgl I'm getting that too now. You can manually edit the file to comment out AVAILABLE_MAC_OS_X_VERSION_10_10_AND_LATER and everything works ok.

I think it has to do with IOKit being 10.10 but CF is still at 10.9. Not a nice fix for it...

**Pull request closed without merging**